lib/stdio/generator.js

```diff
@@ -17,79 +18,13 @@ The `highWaterMark` is kept as the default value, since this is what `childProce
 We ensure `objectMode` is `false` for better buffering.
 
 Chunks are currently processed serially. We could add a `concurrency` option to parallelize in the future.
-
-We return a `Duplex`, created by `Duplex.from()` made of a writable stream and a readable stream, piped to each other.
-- The writable stream is a simple `PassThrough`, so it only forwards data to the readable part.
-- The `PassThrough` is read as an iterable using `passThrough.iterator()`.
-- This iterable is transformed to another iterable, by applying the encoding generators.
-	Those convert the chunk type from `Buffer` to `string | Uint8Array` depending on the encoding option.
-- This new iterable is transformed again to another one, this time by applying the user-supplied generator.
-- Finally, `Readable.from()` is used to convert this final iterable to a `Readable` stream.
 */
-export const generatorToTransformStream = ({value, encoding}) => {
-	const objectMode = false;
-	const highWaterMark = getDefaultHighWaterMark(objectMode);
-	const passThrough = new PassThrough({objectMode, highWaterMark, destroy: destroyPassThrough});
-	const iterable = passThrough.iterator();
-	const encodedIterable = applyEncoding(iterable, encoding);
-	const mappedIterable = value(encodedIterable);
-	const readableStream = Readable.from(mappedIterable, {objectMode, highWaterMark});
-	const duplexStream = Duplex.from({writable: passThrough, readable: readableStream});
+export const generatorToDuplexStream = ({value, encoding}) => {
+	const generators = [getEncodingStartGenerator(encoding), value];
+	const duplexStream = generatorsToDuplex(generators, {objectMode: false});
```

Comment:
Hopefully, this makes this function easier to understand now. 

All we are doing is:
  - Apply the encoding generator, to ensure each chunk is either `string` or `Uint8Array`
  - Apply the user-supplied generator
  - Convert this into a `Duplex` stream